### DISCUSSION TITLE
Recurring warning after using functions that permute data

### DISCUSSION THREAD

**Discussion by Hunterefs** [vegandevs/vegan#482] [Q&A]
Hello,

I've been using vegan to analyze a set of gene expression data - I have lots of columns so I have been using rda() and metaMDS() to permute my dataset. Often after doing so I encounter this warning: 

`x <- rda(data)
Warning messages:
1: Unknown or uninitialised column: `loadings`. 
2: Unknown or uninitialised column: `loadings`.`

Once this warning message appears, it does not disappear until I clear my environment or start a new session. Strangely, R will produce this warning when I attempt to run unrelated code - even comment lines will trigger it. It is also variable in how many times it produces this warning, it can range from 1 to 20+ identical warnings. Reloading a workspace image that this warning occurred in will continue to give warning - it seems like the data needs to be totally cleared. Also, I'm not sure if it is  relevant or not, but my dataset includes negative values.

This is the first time I have submitted a ticket, so please let me know if I can include anything more that might help with this!

Thanks,
- Hunter

**Comment by gavinsimpson** [comment]
It sounds like the output raising the warning is being buffered.

You could turn warnings into errors and then do a `traceback()` when you hit the warning (now error) to see what causing it, and post that `traceback()` output if you need more help tracking it down.

```r
op <- options(warn = 3)
```

If you want to turn this off again, do

```r
options(op)
```

Also make sure you are loading a hidden `.RData` file on startup.

**Comment by Hunterefs** [comment]
Hi Gavin, I tried that `options()` line and the warning stopped popping up so I couldn't trace it back. I'll try to code as much as I can with that `options()` line turned on. 

What do you mean by loading a hidden `.RData` file? Just prior to submitting this ticket I actually went and deleted the hidden `.RData` file my workspace kept loading from, since it would reload whatever was causing the warning.
